subcommand/lifecycle-sidecar/command.go

```diff
@@ -113,7 +115,7 @@ func (c *Command) Run(args []string) int {
 
 // validateFlags validates the flags and returns the parsed syncPeriod and
 // logLevel.
-func (c *Command) validateFlags() (syncPeriod time.Duration, logLevel hclog.Level, err error) {
+func (c *Command) validateFlags() (logLevel hclog.Level, err error) {
```

Comment:
I just realized this, but this is the only place in all of `consul-k8s` that we use named return values. Is there a compelling reason to do so? It doesn't match any of the other code style in the project.

Reply:
I really like the named return values because it makes the code shorter (Go will auto declare and initialize them to zero values). Another nice thing about them is that you can see the meaning of return values. For example, rather than having a function that returns `(string, string)` you could have `(name string, namespace string)`. 

In this case, though it was definitely a gotcha for me because it's easy to overlook the "magic". Plus, having `syncPeriod` be 0 wasn't actually a valid value because then you can't terminate the command at all. 

My overall feeling is that they're good to use but I don't think it makes sense to require that all functions use it in the code.

Reply:
I've also seen a pattern from consul:
```go
	returnErr := func(e error) (time.Duration, hclog.Level, error) {
		return 1 * time.Second, hclog.Info, err
	}
	if c.flagServiceConfig == "" {
		return returnErr(errors.New("-service-config must be set"))
	}
```

It's important to match code style but named returns are pretty idiomatic golang so I think either way is fine, as in we haven't set precedence elsewhere for dealing with this situation (afaik) so I don't think we're going against our code style, more that we're encountering a new situation.

Reply:
I can understand why named return values might be useful, but I don't see that those arguments apply in this case, especially with the gotchas you mention @ishustava . In this case, it's pretty clear what the return values are (hclog.Level and error) and without naming, it would be possible to have a single line return instead of having to define `err` and then return.

I'm happy to revisit the pattern elsewhere when it adds additional value, but this doesn't seem like that place.

Reply:
Sure, I'm happy to remove them.

Since we've started the code style conversation, I realized that parsing and returning logLevel in a validation function is not very common. I've updated it to just return `error`, and when we need to pass it to the logger we can just call the parse function again. It's safe to do because the parsing function does not return an error.